### PR TITLE
Add rust code-generation from proto files

### PR DESCRIPTION
**Pull request by lubkoll**

### CODE REVIEW

File: smart-contracts/proto-build/buf.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+version: v1
+name: buf.build/neutron-org/neutron-sdk
```
Comment (LaurensKubat) [inline smart-contracts/proto-build/buf.yaml:2]: This name should be changed

File: smart-contracts/proto-build/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    let sdk_dir = Path::new(QUASAR_DIR);
+    let proto_path = sdk_dir.join("proto");
+    let proto_paths = [format!("{}/proto/", sdk_dir.display())];
+
+    // List available proto files
+    let mut protos: Vec<PathBuf> = vec![];
+    collect_protos(&proto_paths, &mut protos);
+
+    // Compile all proto client for GRPC services
+    info!("Compiling neutron proto clients for GRPC services!");
```
Comment (LaurensKubat) [inline smart-contracts/proto-build/src/main.rs:119]: neutron can be replaced with Quasar

File: smart-contracts/proto-build/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+    ($fmt:expr, $($arg:tt)+) => {
+        info!(&format!($fmt, $($arg)+))
+    };
+}
+
+fn main() {
+    let tmp_build_dir: PathBuf = TMP_BUILD_DIR.parse().unwrap();
+    let proto_dir: PathBuf = PROTO_DIR.parse().unwrap();
+
+    if tmp_build_dir.exists() {
```
Comment (LaurensKubat) [inline smart-contracts/proto-build/src/main.rs:35]: Should we also cleanup this dir after running main?

File: smart-contracts/packages/quasar-std/src/quasarlabs.quasarnode.tokenfactory.v1beta1.rs [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    #[prost(string, tag = "2")]
+    pub denom: ::prost::alloc::string::String,
+    #[prost(string, tag = "3")]
+    pub new_admin: ::prost::alloc::string::String,
+}
+/// MsgChangeAdminResponse defines the response structure for an executed
+/// MsgChangeAdmin message.
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct MsgChangeAdminResponse {}
+// message MsgForceTransfer {
```
Comment (LaurensKubat) [inline smart-contracts/packages/quasar-std/src/quasarlabs.quasarnode.tokenfactory.v1beta1.rs:144]: Why is this commented

**Review by LaurensKubat — CHANGES_REQUESTED** [review]
Some comments on the naming in the generation code, rest looks good.
Follow up work I do clearly see coming forth from this PR:
- Removing any Osmosis proto definitions from Quasar chain, so we don't generate gamm and lockup code from osmosis
- Expanding our generated code with for example the osmosis-derive or something similar so that quasar-std can be used without a lot of boilerplate
- potentially incorporating quasar-types into quasar-std

File: smart-contracts/proto-build/buf.yaml (hunk 1 above)
Comment (lubkoll) [reply smart-contracts/proto-build/buf.yaml]: done

File: smart-contracts/proto-build/src/main.rs (hunk 2 above)
Comment (lubkoll) [reply smart-contracts/proto-build/src/main.rs]: removed the name

File: smart-contracts/proto-build/src/main.rs (hunk 3 above)
Comment (lubkoll) [reply smart-contracts/proto-build/src/main.rs:35]: yes, done

File: smart-contracts/packages/quasar-std/src/quasarlabs.quasarnode.tokenfactory.v1beta1.rs (hunk 4 above)
Comment (lubkoll) [reply smart-contracts/packages/quasar-std/src/quasarlabs.quasarnode.tokenfactory.v1beta1.rs:144]: because it is commented out in the proto-file

Comment (LaurensKubat) [reply smart-contracts/packages/quasar-std/src/quasarlabs.quasarnode.tokenfactory.v1beta1.rs:144]: makes sense

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Pull request merged**